### ISSUE TITLE
issues when using skipCodeExchange=true

### ISSUE THREAD

**Issue by liatnetach** [FormidableLabs/react-native-app-auth#696]
## Issue
I have 2 weird issues while trying to define skipCodeExchange=true in the configuration file:

1. In **Android**- i'm not getting the state parameter from the authorize response, and in IOS I get it.
what can be the problem and most important how can I fix it?

2. how can I use https redirect link? when Im using non https everything work has should and the browser does closed (both ios and android) f.e. admin://authorize
**but** when Im trying to add https://some.host/somepath link it stack on the web session and doesn't redirect to the app.
both Android and IOS 
**(worth mentioning that every change is made also from the android manifest redirect activity)**
---

## Environment

* **Your Identity Provider**: `vista-print, Azure`
* **Platform that you're experiencing the issue on**: `the first issue is in Android, the second is both`
* **Are you using Expo?**: 'no'

**Comment by Jay-A-McBee** [comment]
:wave: hey @liatnetach - so it looks like the `state` value is [included on the response object in AppAuth-Android](https://github.com/openid/AppAuth-Android/blob/master/library/java/net/openid/appauth/AuthorizationResponse.java#L515). The issue is we don't add this value to [what we send back over the bridge ](https://github.com/FormidableLabs/react-native-app-auth/blob/main/android/src/main/java/com/rnappauth/utils/TokenResponseFactory.java#L74). Let me know if you can work around this. If not, I can talk with the other maintainers about adding this to the response object.

Using an `https` redirect link is more involved. If you haven't done this already, you'll need to create a two way association between your app and the redirect domain. This will allow your app to "claim" `https` url patterns that should open in the app and not in the browser. For iOS this is called a "universal link" - [docs around creating the two association](https://developer.apple.com/documentation/xcode/allowing-apps-and-websites-to-link-to-your-content?preferredLanguage=occ).

On Android you should be able to add the correct scheme to your manifest - [see AppAuth-Android's docs section on capturing the redirect](https://github.com/openid/AppAuth-Android#capturing-the-authorization-redirect) - the "If an HTTPS redirect URI is required instead of a custom scheme" section specifically. Note that Android OS M and above also have a [mechanism](https://developer.android.com/training/app-links) for a two way app/domain association that adds security.

**Comment by liatnetach** [comment]
@Jay-A-McBee I had issues when I used HTTPS.
I defined everything that was needed in both IOS and Android.
but in Android I had an issue that after the first login through the provider the redirect didn't worked.
I read about it in the native Android lib and I saw that it's a common issue, security reasons, cause when the browser try to redirect without any user action it blocks the redirect...
(In IOS it didn't worked also but Im not sure why)
thats why we tried to continue with the costume scheme.

I commented my current issue in the other question so I will close this one.
Thanks.
